qiime2/sdk/action.py

```diff
@@ -284,6 +282,47 @@ def _build_annotations(self):
 
         return annotations
 
+    def _build_numpydoc(self):
+        numpydoc = ""
+        numpydoc += "{}\n\n".format(self.name)
+        numpydoc += "{}\n\n".format(self.description)
+
+        sig = self.signature
+        params = True if len(sig.inputs) != 0 else False
+
+        numpydoc += "{}".format(self._build_section(
+            "Parameters", sig.inputs))
+        numpydoc += "{}".format(self._build_section(
+            "Parameters", sig.parameters, parameters=params))
+        numpydoc += ("\n" if len(sig.parameters) != 0 or
+                     len(sig.inputs) != 0 else "")
+        numpydoc += "{}\n".format(self._build_section(
+            "Returns", sig.outputs))
+
+        return numpydoc
+
+    def _build_section(self, header, iterable, parameters=False):
+        section = ""
+
+        if len(iterable) != 0:
+            if not parameters:
+                section += "{}\n".format(header)
+                section += "{}\n".format(''.join(['-' for char in header]))
+            for item in iterable:
+                item_obj = iterable[item]
+                section += "{}: {}".format(item, item_obj.qiime_type)
```

Comment:
I've never seen format used without keys. I would either add some keys or use string interpolation (`%`). Either is fine.

Also, you should probably use the `repr` call on the QIIME Type, it's closer to what you need semantically (the results are the same at present however). This can be done with `{!r}` or `%r` depending on the interpolation/formatting syntax you go with.